### PR TITLE
get rid of almost all qcommon.h cvar_t

### PR DESCRIPTION
**Pull request by bmorel**
Does not builds yet because I don't know how to fix:

> daemon/src/engine/qcommon/net_ip.cpp:1708:15: error: no member named 'modified' in 'Cvar::Cvar<int>'
        net_enabled->modified = false;

The only remaining cvar_t in that header file will then be `com_version` because I'm even less sure of how to migrate it. Maybe I'll just avoir migrating net_enabled, though. Dunno yet.

### CODE REVIEW

**Comment by slipher** [comment]
FYI for the gamelogic cvar migration I used a [script](https://github.com/slipher/source-tools/blob/master/cvar_migrate.py) to automate the more tedious parts. However the script needs to be modified to work on engine cvars also.

I would recommend skipping net_enabled because of #908. 

Are you going to provide descriptions for the cvars?

**Comment by bmorel** [comment]
> Are you going to provide descriptions for the cvars?

I might do my best for that, but I'm afraid in many cases it will still be a "no" sadly.
I'll look at your script, thanks.

**Comment by bmorel** [comment]
Removed the net_enabled cvar which was incomplete anyway, let's see if things build now.

File: src/engine/qcommon/common.cpp [hunk 1]
```diff
@@ -52,25 +52,27 @@ Maryland 20850 USA.
 #include "sys/sys_events.h"
 #include <common/FileSystem.h>
 
+Cvar::Cvar<bool> com_developer = Cvar::Cvar<bool>( "developer", "", Cvar::TEMPORARY, false );
+Cvar::Cvar<int> com_speeds = Cvar::Cvar<int>( "com_speeds", "", Cvar::NONE, 0 );
```
Comment (necessarily-equal) [inline src/engine/qcommon/common.cpp:56]: I think you can remove the ` = Cvar::Cvar<…>` part of these lines

File: src/engine/qcommon/common.cpp [hunk 2]
```diff
@@ -52,25 +52,27 @@ Maryland 20850 USA.
...
+Cvar::Cvar<float> com_timescale( "timescale", "", Cvar::CHEAT | Cvar::SYSTEMINFO, 1.0f );
+
 cvar_t *com_pid; // bani - process id
 
-cvar_t *com_speeds;
-cvar_t *com_developer;
-cvar_t *com_timescale;
-cvar_t *com_dropsim; // 0.0 to 1.0, simulated packet drops
+Cvar::Range<Cvar::Cvar<float>> com_dropsim = Cvar::Range<Cvar::Cvar<float>>( "com_dropsim", "simulated packet drop", Cvar::CHEAT, 0.0f, 0.0f, 1.0f );
```
Comment (necessarily-equal) [inline src/engine/qcommon/common.cpp:61]: I think you can remove the ` = Cvar::Cvar<…>` part of this line

File: src/engine/qcommon/common.cpp [hunk 3]
```diff
@@ -52,25 +52,27 @@ Maryland 20850 USA.
...
-cvar_t *com_version;
 
-cvar_t *com_unfocused;
-cvar_t *com_minimized;
+cvar_t *com_version;
+Cvar::Cvar<bool> com_sv_running = Cvar::Cvar<bool>( "sv_running", "", Cvar::ROM, false );
+Cvar::Cvar<bool> com_unfocused = Cvar::Cvar<bool>( "com_unfocused", "", Cvar::ROM, false );
+Cvar::Cvar<bool> com_minimized = Cvar::Cvar<bool>( "com_minimized", "", Cvar::ROM, false );
+//NOTE: this one is never used in that file, this may belong somewhere else
+Cvar::Cvar<bool> com_cl_running = Cvar::Cvar<bool>( "cl_running", "", Cvar::ROM, false );
```
Comment (necessarily-equal) [inline src/engine/qcommon/common.cpp:75]: I think you can remove the ` = Cvar::Cvar<bool>` part of these lines

**Review by necessarily-equal — COMMENTED** [review]
Had a quick early read, thanks for starting this.

File: src/engine/qcommon/common.cpp (hunk 1 above)
Comment (bmorel) [reply src/engine/qcommon/common.cpp:56]: nitpick.

File: src/engine/qcommon/common.cpp (hunk 2 above)
Comment (bmorel) [reply src/engine/qcommon/common.cpp:61]: nitpick.

File: src/engine/qcommon/common.cpp (hunk 3 above)
Comment (bmorel) [reply src/engine/qcommon/common.cpp:75]: nitpick.

**Pull request closed without merging**